sideboard/config.py

```diff
@@ -13,6 +13,16 @@ class ConfigurationError(RuntimeError):
     pass
 
 
+def get_config_root():
+    """
+    Returns the configuration root for the system, which defaults to '/etc/sideboard'.
+    If the SIDEBOARD_CONFIG_ROOT environment variable is set, its contents will be returned instead.
+    """
+    config_root = os.environ.get('SIDEBOARD_CONFIG_ROOT', '/etc/sideboard')
+    assert os.path.isdir(config_root) and os.access(config_root, os.R_OK), '{!r} directory is not readable'.format(config_root)
```

Comment:
Ooh, good catch.  This happens to have not bitten us at my day job because of how we provision our Vagrant environments, but it would totally bite anyone who happened to provision their Vagrant boxes without actually installing Sideboard!

The problem with logging a warning is that we haven't actually configured logging yet.  So how about this:
- if the ``SIDEBOARD_CONFIG_ROOT`` environment variable is NOT set, then we don't do anything if ``/etc/sideboard`` doesn't exist because that's a totally valid and expected use case
- if the directory DOES exist but we can't read it, then we treat that as an error and raise an exception
- if the ``SIDEBOARD_CONFIG_ROOT`` environment variable IS set, then we raise an exception if it doesn't exist or we can't read it

That behavior seems intuitive to me; if you haven't set it then you might not care, but if you do set it then we'd expect it to be valid.

How does that sound?